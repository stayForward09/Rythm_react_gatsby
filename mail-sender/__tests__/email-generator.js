// Copyright (c) 2018-2019 AndreaSonny <andreasonny83@gmail.com> (https://github.com/andreasonny83)
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

const { EmailGenerator } = require('../src/email-generator');

describe('EmailGenerator', () => {
  beforeEach(() => (emailGenerator = new EmailGenerator()));

  it('should export a function', () => {
    // Assert
    expect(typeof EmailGenerator).toEqual('function');
  });

  it('should set the charset to `UTF-8` by default', () => {
    // Arrange
    const expectedCharset = 'UTF-8';

    // Act
    const result = emailGenerator.generate();

    // Assert
    expect(result.Message.Subject.Charset).toEqual(expectedCharset);
  });

  it('should accept a different charset', () => {
    // Arrange
    const expectedCharset = 'ISO-8859-1';

    // Act
    const isoEmailGenerator = new EmailGenerator(expectedCharset);
    const result = isoEmailGenerator.generate();

    // Assert
    expect(result.Message.Subject.Charset).toEqual(expectedCharset);
  });

  it('should generate an email object', () => {
    // Arrange
    const emailGenerator = new EmailGenerator();
    const subject = 'test email';
    const branch = 'Earth';
    const from = 'fake_user@mail.com';
    const sender = 'Heisenberg';
    const email = 'heisenberg@crystal.blue';
    const to = 'fake_recipient@mail.com';
    const body = 'Hello!';
    // Mock the Date constructor
    Date = jest.fn(() => ({
      toUTCString: () => 'now',
    }));

    // Act
    const prepareMail = emailGenerator.generate(
      subject,
      body,
      branch,
      from,
      to,
      sender,
      email
    );

    // Assert
    expect(prepareMail).toMatchSnapshot();
  });
});
