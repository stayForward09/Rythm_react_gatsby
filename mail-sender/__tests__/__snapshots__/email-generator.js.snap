// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailGenerator should generate an email object 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "fake_recipient@mail.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
<html><head><title>RhythmicExcellence.london</title></head>
  <body leftmargin=\\"0\\" marginwidth=\\"0\\" topmargin=\\"0\\" marginheight=\\"0\\" offset=\\"0\\">
    <center>
      <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" height=\\"100%\\" width=\\"100%\\" id=\\"bodyTable\\">
        <tr><td align=\\"center\\" valign=\\"top\\" style=\\"padding-bottom:40px;\\">
          <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"600\\" id=\\"templateContainer\\">
            <tr><td align=\\"center\\" valign=\\"top\\" style=\\"padding-bottom:20px;\\">
              <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"600\\" id=\\"templatePreheader\\"></table>
            </td></tr>
            <tr><td align=\\"center\\" valign=\\"top\\">
              <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"600\\" id=\\"templateHeader\\">
                <tr><td align=\\"center\\" valign=\\"top\\" style=\\"padding-top:20px; padding-bottom:20px;\\">
                  <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                    <tr><td class=\\"headerContent\\" mc:edit=\\"header_content\\">
                      <h1>Hi RhythmicExcellence,<br />
                        <span style=\\"font-size:30px\\">Heisenberg has just left you a message:</span>
                      </h1>
                      <br />
                      Date: now<br/>
                      From: heisenberg@crystal.blue<br/>
                      Branch: Earth<br/>
                    </td></tr>
                  </table>
                </td></tr>
              </table>
            </td></tr>
            <tr><td align=\\"center\\" valign=\\"top\\">
              <table border=\\"0\\" cellpadding=\\"40\\" cellspacing=\\"0\\" width=\\"600\\" id=\\"templateBody\\">
                <tr><td align=\\"center\\" valign=\\"top\\" style=\\"padding-top:20px; padding-bottom:20px;\\">
                  <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                    <tr><td valign=\\"top\\" class=\\"upperBodyContent\\" mc:edit=\\"body_content01\\">
                      Hello!
                    </td></tr>
                  </table>
                </td></tr>
              </table>
            </td></tr>
            <tr><td align=\\"center\\" valign=\\"top\\">
              <table border=\\"0\\" cellpadding=\\"20\\" cellspacing=\\"0\\" width=\\"600\\" id=\\"templateFooter\\">
                <tr><td align=\\"center\\" valign=\\"top\\" style=\\"padding-right:40px; padding-left:40px;\\">
                  <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                    <tr><td valign=\\"top\\" class=\\"footerContent\\" style=\\"border-top:1px solid #BBBBBB; padding-top:20px;\\" mc:edit=\\"footer_social\\"></td></tr>
                    <tr><td valign=\\"top\\" class=\\"footerContent\\" style=\\"padding-top:20px;\\" mc:edit=\\"footer_utility\\">
                      
    This is an automatic e-mail generated from <a href=\\"http://www.rhythmicexcellence.london\\" target=\\"_blank\\">www.rhythmicexcellence.london</a><br/>
    If you do not wish to continue receiving these messages or for other queries, please contact your web administrator at: <a href=\\"mailto:andreasonny83@gmail.com\\" target=\\"_blank\\">andreasonny83@gmail.com</a>.
                    </td></tr>
                  </table>
                </td></tr>
              </table>
            </td></tr>
          </table>
        </td></tr>
      </table>
    </center>
  </body>
</html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
      From: heisenberg@crystal.blue <Heisenberg> <br/>
      Date: now <br/>
      Branch: Earth <br/>
      Message: Hello! <br/>
      Footer: 
    This is an automatic e-mail generated from http://rhythmicexcellence.london<br/>
    If you do not wish to continue receiving these messages or for other queries, please contact your web administrator at: andreasonny83@gmail.com . <br/>
      ",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "test email",
    },
  },
  "Source": "fake_user@mail.com",
}
`;
